src/main/java/edu/ie3/datamodel/io/source/IdCoordinateSource.java

```diff
@@ -5,14 +5,57 @@
 */
 package edu.ie3.datamodel.io.source;
 
+import edu.ie3.datamodel.utils.CoordinateDistance;
+import edu.ie3.datamodel.utils.GridAndGeoUtils;
 import java.util.Collection;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.stream.Collectors;
 import org.locationtech.jts.geom.Point;
 
 public interface IdCoordinateSource extends DataSource {
 
+  /**
+   * Get the matching coordinate for the given ID
+   *
+   * @param id the ID to look up
+   * @return matching coordinate
+   */
   Point getCoordinate(Integer id);
 
+  /**
+   * Get the matching coordinates for the given IDs
+   *
+   * @param ids the IDs to look up
+   * @return the matching coordinates
+   */
   Collection<Point> getCoordinates(Integer... ids);
 
+  /**
+   * Get the ID for the coordinate point
+   *
+   * @param coordinate the coordinate to look up
+   * @return the matching ID
+   */
   Integer getId(Point coordinate);
+
+  /**
+   * Returns all the coordinates of this source
+   *
+   * @return all available coordinates
+   */
+  Collection<Point> getAllCoordinates();
+
+  /**
+   * Returns the nearest n coordinate points to the given coordinate
+   *
+   * @param coordinate the coordinate to look up the nearest neighbours for
+   * @param n how many neighbours to look up
+   * @return the n nearest coordinates to the given point
+   */
+  default List<CoordinateDistance> getNearestCoordinates(Point coordinate, Integer n) {
+    SortedSet<CoordinateDistance> sortedDistances =
+        GridAndGeoUtils.getCoordinateDistances(coordinate, getAllCoordinates());
```

Comment:
I'm a little bit afraid that this could become a performance bottleneck if you need to query `getAllCoordinates()` every time. Is this really necessary? Maybe you could consider providing something like the example below. This would allow passing in already queried coordinates. Does this makes sense in your scenario? If you take this solution pls add and adapt javadocs accordingly. 

```java
    default List<CoordinateDistance> getNearestCoordinates(Point coordinate, int n) {
        return getNearestCoordinates(coordinate, n, Collections.emptySet());
    }

    default List<CoordinateDistance> getNearestCoordinates(Point coordinate, int n, Collection<Point> allCoordinates) {
        SortedSet<CoordinateDistance> sortedDistances = GridAndGeoUtils.getCoordinateDistances(coordinate,
                        (allCoordinates == null || allCoordinates.isEmpty()) ? getAllCoordinates() : allCoordinates);
        return sortedDistances.stream().limit(n).collect(Collectors.toList());
    }
}
```